lib/utils.js

```diff
@@ -93,6 +93,10 @@ exports.deepEqual = function deepEqual(a, b) {
     return a.toString() === b.toString();
   }
 
+  if (isBsonType(a, 'Int32') && isBsonType(b, 'In32')) {
+    return a.value === b.value();
+  }
```

Comment:
We don't need this change, deepEquality already worked for Int32 classes.  But if we do leave it in this PR:

```suggestion
  if (isBsonType(a, 'Int32') && isBsonType(b, 'Int32')) {
    return a.value === b.value;
  }
```

Reply:
I removed the change, good catch!